Docs/Deploy_YAMS.md

```diff
@@ -5,66 +5,158 @@ This tutorial will show you how to configure YAMS and deploy it to a cloud servi
+              RunAsync().Wait();
+          }
+
+          public async Task RunAsync()
+          {
+              WorkerRoleConfig config = new WorkerRoleConfig();
+              YamsConfig yamsConfig = new YamsConfigBuilder(
+                  // mandatory configs
+                  DeploymentIdUtils.GetYamsClusterId(this.IsSingleClusterDeployment),
+                  RoleEnvironment.CurrentRoleInstance.UpdateDomain.ToString(),
+                  RoleEnvironment.CurrentRoleInstance.Id,
+                  config.CurrentRoleInstanceLocalStoreDirectory)
+                  // optional configs
+                  .SetCheckForUpdatesPeriodInSeconds(config.UpdateFrequencyInSeconds)
+                  .SetApplicationRestartCount(config.ApplicationRestartCount)
+                  .Build();
+              _yamsService = YamsServiceFactory.Create(yamsConfig,
+                  deploymentRepositoryStorageConnectionString: config.StorageDataConnectionString,
+                  updateSessionStorageConnectionString: config.StorageDataConnectionString);
+
+              try
+              {
+                  Trace.TraceInformation("Yams is starting");
+                  await _yamsService.Start();
+                  Trace.TraceInformation("Yams has started. Looking for apps with deploymentId:" + yamsConfig.ClusterDeploymentId);
+                  while (true)
+                  {
+                      await Task.Delay(1000);
+                  }
+              }
+              catch (Exception ex)
+              {
+                  Trace.TraceError(ex.ToString());
+              }
+          }
+
+          public override bool OnStart()
+          {
+              Trace.TraceInformation("Yams WorkerRole is starting");
+              ServicePointManager.DefaultConnectionLimit = 1000;
+              RoleEnvironment.Changing += RoleEnvironmentChanging;
+              var result = base.OnStart();
+              Trace.TraceInformation("Yams WorkerRole has started");
+              return result;
+          }
+
+          public override void OnStop()
+          {
+              StopAsync().Wait();
+          }
+
+          public async Task StopAsync()
+          {
+              Trace.TraceInformation("Yams WorkerRole is stopping");
+              RoleEnvironment.Changing -= RoleEnvironmentChanging;
+              if (_yamsService != null)
+              {
+                  await _yamsService.Stop();
+              }
+              base.OnStop();
+              Trace.TraceInformation("Yams has stopped");
+          }
+
+          private void RoleEnvironmentChanging(object sender, RoleEnvironmentChangingEventArgs e)
+          {
+              // If a configuration setting is changing);
+              if (e.Changes.Any(change => change is RoleEnvironmentConfigurationSettingChange))
+              {
+                  e.Cancel = true;
+              }
+          }
+      }
+  ```
+
+You can create additional Worker Roles that inherit from this Worker Role.
```

Comment:
This is a bit dangerous because users might add a second worker role and end up with conflicting Yams cluster id. I'd either explain that concatenation would be needed as done in the multiple clusters sample or remove this sentence.